python/dask_cudf/dask_cudf/tests/test_core.py

```diff
@@ -32,6 +32,32 @@ def test_from_dict_backend_dispatch():
     dd.assert_eq(expect, ddf)
 
 
+@xfail_dask_expr("API collision with dask-expr")
```

Comment:
IIUC, this fails because we successfully raise the `FutureWarnings` around using `to/from_dask_dataframe`, but that call doesn't actually work as expected if query-planning is enabled and so we don't actually update the `_meta` object?

If this is the case, could we modify these tests so that we have:

- tests for just the emitting of the warning (which should pass with or without query planning)
- tests for the successful use of the deprecated function (which should only work with query planning disabled)

Reply:
This is partially correct. When query-planning is enabled, we end up using the dask-expr API for `to/from_dask_dataframe`. We don't use the `dask_cudf/core.py` code path at all, and don't throw any future warnings.

I'm now realizing that we really have no choice but to override the dask-expr API in dask-cudf - Will need to revise this.